jwst/assign_wcs/miri.py

```diff
@@ -134,102 +136,212 @@ def imaging_distortion(input_model, reference_files):
+    # bb_sub will be the bounding box in subarray coordinates
+    # bb will be the bounding box in full-array coordinates
     subarray_xstart = input_model.meta.subarray.xstart - 1
     subarray_ystart = input_model.meta.subarray.ystart - 1
-
-    # The bounding box is computed from the data in the wavelength solution
-    # and corrected for subarray.
-    bb = ((x0.min() - 0.5 + zero_point[0] - subarray_xstart, x1.max() + 0.5 + zero_point[0] - subarray_xstart),
-          (y0.min() - 0.5 + zero_point[1] - subarray_ystart, y0.max() + 0.5 + zero_point[1] - subarray_ystart))
-
-    # Compute the v2v3 to sky.
-    tel2sky = pointing.v23tosky(input_model)
-
-    # To compute the spatial detector to V2V3 transform:
-    # Take a row centered on zero_point_y and convert it to v2, v3.
-    # The forward transform uses constant ``y`` values for each ``x``.
-    # The inverse transform uses constant ``v3`` values for each ``v2``.
-    zero_point_v2v3 = distortion(*zero_point)
-
-    spatial_forward = models.Identity(1) & models.Const1D(zero_point[1]) | distortion
-    spatial_forward.inverse = (models.Identity(1) & models.Const1D(zero_point_v2v3[1]) |
-                               distortion.inverse)
-
-    # Create the spectral transforms.
-    lrs_wav_model = jwmodels.LRSWavelength(lrsdata, zero_point)
-
+    
+    # If in fixed slit mode, define the bounding box using the corner locations provided in
+    # the CDP reference file.
+    if input_model.meta.exposure.type.lower() == 'mir_lrs-fixedslit':
+        
+        bb_sub = ((np.floor(x0.min()+ zero_point[0]) - 0.5, np.ceil(x1.max()+ zero_point[0]) + 0.5),
+                  (np.floor(y2.min()+ zero_point[1]) - 0.5, np.ceil(y0.max()+ zero_point[1]) + 0.5))
+        bb = ((bb_sub[0][0]+subarray_xstart,bb_sub[0][1]+subarray_xstart),
+              (bb_sub[1][0]+subarray_ystart,bb_sub[1][1]+subarray_ystart))
+        
+    # If in slitless mode, define the bounding box X locations using the subarray x boundaries
+    # and the y locations using the corner locations in the CDP reference file.  Make sure to
+    # omit the 4 reference pixels on the left edge of slitless subarray.
+    if input_model.meta.exposure.type.lower() == 'mir_lrs-slitless':
+        bb_sub = ((input_model.meta.subarray.xstart-1+4-0.5,input_model.meta.subarray.xsize-1+0.5),
+                  (np.floor(y2.min()+ zero_point[1]) - 0.5, np.ceil(y0.max()+ zero_point[1]) + 0.5))
+        bb = ((bb_sub[0][0]+subarray_xstart,bb_sub[0][1]+subarray_xstart),
+              (bb_sub[1][0]+subarray_ystart,bb_sub[1][1]+subarray_ystart))
+
+    # Find the ROW of the zero point
+    row_zero_point = zero_point[1]
+    # Make a vector of x,y locations for every pixel in the reference row
+    yrow, xrow = np.mgrid[row_zero_point:row_zero_point + 1, 0:input_model.data.shape[1]]
+    # And compute the v2,v3 coordinates of pixels in this reference row
+    v2v3refrow = np.array(subarray_dist(xrow, yrow))[:, 0, :]
```

Comment:
I am assuming the zero point for slitless mode is in subarray coordinates (which it seems to be) or this won't be correct.

Reply:
Right; as above the zeropoint is converted to subarray coordinates when first read from the reference file.